### DISCUSSION TITLE
Are integer matrices supported in cglm ?

### DISCUSSION THREAD

**Discussion by SeleDreams** [recp/cglm#351] [Q&A]
I am working on a C game engine for some very low end platforms (such as the DS) where fixed point is the most efficient way to implement 3d, as such I'd need the matrices etc to use fixed point values.
does cglm have any way to handle those ? it seems like the standard glm does have integer matrices but i don't really see them in cglm
https://glm.g-truc.net/0.9.5/api/a00174.html

**Comment by recp** [comment]
Hi @SeleDreams,

Currently int matrices are not implemented yet, but can be asap ;) May I ask which integer size do you need in your engine? 32 or lower? Standart `imat` matrices can be implemented in **cglm** 

FWIW, for lower bits, a PR is still waiting: https://github.com/recp/cglm/pull/348 for int vectors.

**Reply by SeleDreams** [reply]
the platforms I plan to support do use 32-bit so yeah, 32-bit is sufficient for my needs

**Reply by recp** [reply]
Great then! It can be implemented faster in this case, I added to my TODOs but can be implemented sooner by contributions ;)
